docs/api/dataset.rst

```diff
@@ -0,0 +1,99 @@
+Datasets
+========
+
+General datasets
+----------------
+
+.. automodule:: fuel.datasets.base
+    :members:
+    :undoc-members:
+    :show-inheritance:
+
+Adult dataset
+-------------
+
+.. automodule:: fuel.datasets.adult
+    :members:
+    :undoc-members:
+    :show-inheritance:
+
+billion dataset
+---------------
+
+.. automodule:: fuel.datasets.billion
+    :members:
+    :undoc-members:
+    :show-inheritance:
+
+CalTech 101 dataset
+-------------------
+
+.. automodule:: fuel.datasets.caltech101_silhouettes
+    :members:
+    :undoc-members:
+    :show-inheritance:
+
+Binarized MNIST dataset
+-----------------------
+
+.. automodule:: fuel.datasets.binarized_mnist
+    :members:
+    :undoc-members:
+    :show-inheritance:
+
+CIFAR100 dataset
+----------------
+
+.. automodule:: fuel.datasets.cifar100
+    :members:
+    :undoc-members:
+    :show-inheritance:
+
+CIFAR10 dataset
+---------------
+
+.. automodule:: fuel.datasets.cifar10
+    :members:
+    :undoc-members:
+    :show-inheritance:
+
+iris dataset
+------------
+
+.. automodule:: fuel.datasets.iris
+    :members:
+    :undoc-members:
+    :show-inheritance:
+
+MNIST dataset
+-------------
+
+.. automodule:: fuel.datasets.mnist
+    :members:
+    :undoc-members:
+    :show-inheritance:
+
+SVHN dataset
+------------
+
+.. automodule:: fuel.datasets.svhn
+    :members:
+    :undoc-members:
+    :show-inheritance:
+
+text dataset
```

Comment:
Maybe `Text-based datasets`? In this case, having `datasets` in the title makes sence I think.